### DISCUSSION TITLE
Firebase authentication using React app failing with 'Cannot read property 'prototype' of undefined'

### DISCUSSION THREAD

**Discussion by sponesta-root** [firebase/flutterfire#11391] [Q&A]
Hi,

I am using Firebase for SigninWithPhone functionality.

Following is code snippet

import React, { useState } from 'react';
import {View, Text, TextInput, Button} from 'react-native'
import firebase from 'firebase/app';
import auth from './firebaseConfig'
import RecaptchaVerifier from 'firebase/auth'

import styles from './assets/styles'
 
const Login = () => {
    // Inputs
    const [mynumber, setnumber] = useState("");
    const [otp, setotp] = useState('');
    const [show, setshow] = useState(false);
    const [final, setfinal] = useState('');
 
    // Sent OTP
    const signin = () => {
        if (mynumber === "" || mynumber.length < 10) return;
           window.RecaptchaVerifier  = new RecaptchaVerifier(auth, 'login-button',{
            size: 'invisible',
            callback: function (response) {
                console.log('It works!');
              },
           });
                const appVerifier = window.RecaptchaVerifier
            signInWithPhoneNumber(auth, mynumber, appVerifier).then((result) => {
                setfinal(result);
                alert("code sent")
                setshow(true);}).catch((err) => {
                    alert(err);
                    window.location.reload()
                });
    }

```
 return (
        <View style={styles.containerCardItem}>
            <TextInput placeholder='Number' onChangeText={newText => setnumber(newText)} />
            <Button id = 'login-button' onPress={signin} title="SendOTP"/>
            <TextInput placeholder='OTP' onChangeText={newText => setotp(newText)}/>
            <Button onPress={ValidateOtp} title="Verify"/> 
        </View>
    );

```

Below is the error
TypeError: Cannot read property 'prototype' of undefined, js engine: hermes
at node_modules\react-native\Libraries\Core\ExceptionsManager.js:null in reportException
at node_modules\react-native\Libraries\Core\ExceptionsManager.js:null in handleException
- ... 7 more stack frames from framework internals

Firebase version used in package Json "firebase": "^10.1.0",

Using expo to run this on mobile

**Comment by ishubs** [comment]
has anyone able to crack this ?
